src/java/grl/src/grl/driver/GRL_Driver.java

```diff
@@ -232,7 +262,9 @@ public void run()
 		while (!stop && !_startStopUI.is_stopped() && _lbr.getSafetyState().getEmergencyStopInt()==EmergencyStop.INACTIVE) {
 			message_counter+=1;
 			_currentKUKAiiwaState = udpMan.waitForNextMessage();
-			_previousKUKAiiwaState = udpMan.getPrevMessage();
+			
+			//Some bug in this, just set _previousKUKAiiwaState = _currentKUKAiiwaState before looping/continue?
+			_previousKUKAiiwaState = udpMan.getPrevMessage();  
```

Comment:
Isee
https://github.com/tdinesh/grl/blob/devel_control_modes/src/java/grl/src/grl/driver/GRL_Driver.java#L521 #L535 should use appropriate controlMode based on the current one. 

The bug is shows up at
https://github.com/tdinesh/grl/blob/devel_control_modes/src/java/grl/src/grl/driver/GRL_Driver.java#L429  

grl.flatbuffer.EControlMode are the same for both _previousKUKAiiwaState & _currentKUKAiiwaState. 

Maybe also the issue for switching 
_currentKUKAiiwaState.armControlState().stateType()